account-kit/privy-integration/src/types.ts

```diff
@@ -0,0 +1,209 @@
+  /** Recipient address */
+  to: Address;
+
+  /** Transaction data (calldata) */
+  data?: Hex;
+
+  /** Transaction value - accepts string | number | bigint */
+  value?: string | number | bigint;
+}
+
+/**
+ * Options for sending a transaction
+ */
+export interface SendTransactionOptions {
+  /**
+   * Whether to sponsor the transaction
+   * Default: true if policy ID exists and defaultSponsored is not set to false
+   */
+  sponsored?: boolean;
+}
+
+/**
+ * Result of a successful transaction
+ */
+export interface SendTransactionResult {
+  /** EVM transaction hash (first receipt hash) */
+  txnHash: Hash;
+}
+
+/**
+ * Hook result for sending transactions
+ */
+export interface UseSendTransactionResult {
+  /** Whether the transaction is currently being sent */
+  isLoading: boolean;
+
+  /** Error if transaction failed */
+  error: Error | null;
+
+  /** Transaction result if successful */
+  data: SendTransactionResult | null;
+
+  /** Reset the hook state */
+  reset(): void;
+
+  /** Send a transaction */
+  sendTransaction(
+    input: UnsignedTransactionRequest,
+    options?: SendTransactionOptions,
+  ): Promise<SendTransactionResult>;
+}
+
+/**
+ * Request parameters for preparing a swap
+ *
+ * Note: Provide either `fromAmount` OR `minimumToAmount`, not both.
+ * - Use `fromAmount` to specify exact amount to swap FROM
+ * - Use `minimumToAmount` to specify minimum amount to receive TO
+ */
+export interface PrepareSwapRequest {
+  /** Address initiating the swap */
+  from?: Address;
+
+  /** Token address to swap from (use "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE" for native token) */
+  fromToken: Address;
+
+  /** Token address to swap to (use "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE" for native token) */
+  toToken: Address;
+
+  /** Exact amount to swap from in hex format (e.g., "0xde0b6b3a7640000" for 1 ETH in wei). Mutually exclusive with minimumToAmount. */
+  fromAmount?: Hex;
+
+  /** Minimum amount to receive in hex format (e.g., "0xde0b6b3a7640000" for 1 ETH in wei). Mutually exclusive with fromAmount. */
+  minimumToAmount?: Hex;
+}
+
+/**
+ * Swap quote information from Alchemy's swap API
+ */
+export interface SwapQuote {
+  /** Amount being swapped from (hex string) */
+  fromAmount: Hex;
+
+  /** Minimum amount to receive (hex string) */
+  minimumToAmount: Hex;
+
+  /** Quote expiration timestamp as hex string (convert with parseInt(expiry, 16)) */
+  expiry: Hex;
+
+  /** Additional quote details */
+  [key: string]: unknown;
+}
+
+/**
+ * Prepared swap calls ready for signing
+ * This matches the structure returned by requestQuoteV0
+ */
+export interface PreparedSwapCalls {
+  /** Swap quote details */
+  quote?: SwapQuote;
+
+  /** Chain ID */
+  chainId?: Hex;
+
+  /** Call type */
+  type?: string;
+
+  /** Not raw calls */
+  rawCalls?: false;
+
+  /** Additional call data */
+  [key: string]: unknown;
+}
+
+/**
+ * Result of preparing a swap
+ */
+export interface PrepareSwapResult {
```

Comment:
Is the quote supposed to be here and in the preparedSwapCalls at the same time? Just wondering

Reply:
Yeah good catch - this type file was from an early prototype I made and forgot to update.  I'm going to update this to use the types from our sdk